src/main/java/com/artipie/asto/fs/FileStorage.java

```diff
@@ -290,6 +299,30 @@ private Path path(final Key key) {
         return Paths.get(this.dir.toString(), key.string());
     }
 
+    /**
+     * Removes empty key parts (directories).
+     * @param key Key
+     */
+    private void deleteEmptyParts(final Optional<Key> key) {
+        try {
+            if (key.isPresent() && !key.get().string().isEmpty()
+                && Files.isDirectory(this.path(key.get()))) {
```

Comment:
@olenagerasimova `this.path(key.get())` calls are used 3 times in this method, it could moved to a variable